src/main/java/moriyashiine/aylyth/mixin/LivingEntityMixin.java

```diff
@@ -63,15 +156,18 @@ private void aylyth_injectLootDrop(CallbackInfo ci){
 		}
 	}
 
+
+	private static final StatusEffect[] persistentStatusEffects = { ModPotions.CRIMSON_CURSE_EFFECT,
+			StatusEffects.WITHER, StatusEffects.INSTANT_DAMAGE, StatusEffects.INSTANT_HEALTH };
+
 	@Inject(method = "canHaveStatusEffect", at = @At("HEAD"), cancellable = true)
 	public void aylyth_canHaveStatusEffect(StatusEffectInstance effect, CallbackInfoReturnable<Boolean> cir) {
-		if (this.isPlayer()) {
-			LivingEntity entity = ((LivingEntity) (Object) this);
-			if(entity instanceof PlayerEntity && ((PlayerEntity) entity).getInventory().contains(ModItems.YMPE_EFFIGY_ITEM.getDefaultStack())) {
-				cir.setReturnValue(effect.getEffectType() == StatusEffects.WITHER ||
-						effect.getEffectType() == StatusEffects.INSTANT_DAMAGE ||
-						effect.getEffectType() == StatusEffects.INSTANT_HEALTH);
-			}
+        if (!this.isPlayer()) {
+            return;
 		}
+
+        LivingEntity entity = ((LivingEntity) (Object) this);
+        if(entity instanceof PlayerEntity && ((PlayerEntity) entity).getInventory().contains(ModItems.YMPE_EFFIGY_ITEM.getDefaultStack()))
+            Arrays.stream(persistentStatusEffects).filter(entity::hasStatusEffect).map(persistentStatusEffect -> false).forEach(cir::setReturnValue);
```

Comment:
This is a behavior change. Using a tag for the aforementioned persistent effects and after the player and effigy checks, check the passed `effect` arg whether it is tagged or not, returning true if it is.

Reply:
Sort of unsure on this, will push a commit with changes for it soon

Reply:
the change lgtm!

Reply:
aight thx